types/generated/database.ts

```diff
@@ -0,0 +1,166 @@
+import type { ColumnType } from "kysely";
+
+export type Generated<T> = T extends ColumnType<infer S, infer I, infer U>
+  ? ColumnType<S, I | undefined, U>
+  : ColumnType<T, T | undefined, T>;
+
+export type Json = ColumnType<JsonValue, string, string>;
+
+export type JsonArray = JsonValue[];
+
+export type JsonObject = {
+  [K in string]?: JsonValue;
+};
+
+export type JsonPrimitive = boolean | null | number | string;
+
+export type JsonValue = JsonArray | JsonObject | JsonPrimitive;
+
+export type Numeric = ColumnType<string, string | number, string | number>;
+
+export type Timestamp = ColumnType<Date, Date | string, Date | string>;
+
+export interface Agencies {
+  id: Generated<string>;
+  name: string;
+  website: string;
+  phone: string;
+  bio: string | null;
+  address_line_1: string;
+  address_line_2: string | null;
+  city: string;
+  state: string;
+  country: string;
+  zip_code: string;
+  verified: Generated<boolean>;
+  employer_identification_number: string | null;
+  account_manager_id: string;
+  image_id: string;
+  created_at: Generated<Timestamp>;
+  updated_at: Timestamp | null;
```

Comment:
```suggestion
  updated_at: Generated<Timestamp> | null;
```

I'm still trying to understand the point behind the `Generated` utility type, but wouldn't the `update_at` Timestamp also be considered `Generated`?

Reply:
I would not touch it since it is auto-generated and messing with it could potentially brick our typing system for the database

My guess as to why it's not auto-generated is that the type generation CLI does not take triggers into account

And technically you can still update that field manually

I generated that file into it's own output to see what we have, usually you would not do that

Reply:
@Enubia Didn't realize the file was generated initially, it not taking triggers into account is most definitely the issue there.

We will see how the types look once we are using them and can make changes if necessary later on.